### DISCUSSION TITLE
Crop Streamlines at a GMWMI

### DISCUSSION THREAD

**Discussion by smeisler** [dipy/dipy#2638] [Tractography / Tractometry]
Hello,

I have a set of streamlines (.tck) for a tract and a binary gray-matter-white-matter-interface (GMWMI). Is there an API in Dipy that I can use to manipulate already-made streamline objects so they do not go beyond the GMWMI? Due to the software used for tract segmentation, I am not sure if this is something that I can incorporate into the tracking itself (similar to how MRTrix has the `--crop-at-gmwmi` option).

For demonstration, I have attached the end of a left arcuate and same subject's GMWMI. You can see there are some streamlines that pass and even leave the GMWMI, and I would like to just alter those streamlines, rather than remove them altogether.
<img width="212" alt="image" src="https://user-images.githubusercontent.com/27028726/187287018-d7ed5f95-7410-44ab-a167-ea7f91edb441.png">

Best,
Steven

**Comment by skoudoro** [comment]
Hi @smeisler, 

I might be wrong, but you can try the function [`near_roi`](https://dipy.org/documentation/1.4.1./reference/dipy.tracking/#near-roi).

I think there is a tutorial for that. I need to look into it. 

Can you confirm @arokem or @frheault?

**Reply by skoudoro** [reply]
also @DanNBullock?

**Reply by smeisler** [reply]
Hi @skoudoro ,

Thanks for the answer. It looks like that would be a method for selecting streamlines based on their termination points. I am looking for something that would modify the streamlines such that they do not pass through a given mask.

**Comment by skoudoro** [comment]
ok, so, the function you are looking for is `target` (a strange name by the way...).  

you can find how to use it in this tutorial: https://dipy.org/documentation/1.5.0/examples_built/streamline_tools/#example-streamline-tools

Just search for `target` in the tutorial above and we explain it in detail. Let me know if there is an issue.
